src/service/routing/core.clj

```diff
@@ -1,6 +1,41 @@
 (ns service.routing.core
-  (:require [compojure.api.sweet :refer :all]
-            [service.routing.spec]))
+  (:require [clojure.spec.gen.alpha :as gen]
+            [ring.util.http-response :refer [ok]]
+            [compojure.api.sweet :refer [context GET api]]
+            [service.routing.spec :as s]
+            [service.routing.directions :as dir]
+            [service.routing.graph.generators :as g]))
+            ;[expound.alpha :as expound]
+
+;(s/valid? (s/and ::radiuses-regex #(= (count coordinates) (count %)))))
+;(s/and ::s/radiuses-regex #(= (count coordinates) (count %))))
+
+(def routes
+  (context "/spec" []
+    :tags ["spec"]
+    :coercion :spec
+
+    (GET "/direction/:coordinates" []
+      :summary "direction with clojure.spec"
+      :path-params [coordinates :- ::s/coordinate-regex]
```

Comment:
@mehdisadeghi it seems that I was wrong :(
According to this blog, the way that you had it was correct: https://kotka.de/blog/2010/05/Did_you_know_III.html
I think that the error that you are getting is probaly due to no spec being registered for `:s/coordinate-regex` but rather for `::s/coordinate-regex`

Although I did tell you that I was not sure and that you should check in the REPL ;)